lib/charms/opensearch/v0/opensearch_relation_peer_cluster.py

```diff
@@ -433,24 +443,25 @@ def _rel_err_data(  # noqa: C901
             blocked_msg = (
                 "Cannot have 2 'failover'-orchestrators. Relate to the existing failover."
             )
-        elif not self.charm.is_admin_user_configured():
-            blocked_msg = f"Admin user not fully configured {message_suffix}."
-        elif not self.charm.tls.is_fully_configured_in_cluster():
-            blocked_msg = f"TLS not fully configured {message_suffix}."
-            should_retry = False
-        elif not self.charm.peers_data.get(Scope.APP, "security_index_initialised", False):
-            blocked_msg = f"Security index not initialized {message_suffix}."
-        elif not self.charm.is_every_unit_marked_as_started():
-            blocked_msg = f"Waiting for every unit {message_suffix} to start."
-        elif not self.charm.secrets.get(Scope.APP, self.charm.secrets.password_key(COSUser)):
-            blocked_msg = f"'{COSUser}' user not created yet."
-        else:
-            try:
-                if not self._fetch_local_cm_nodes(deployment_desc):
-                    blocked_msg = f"No 'cluster_manager' eligible nodes found {message_suffix}"
-            except OpenSearchHttpError as e:
-                logger.error(e)
-                blocked_msg = f"Could not fetch nodes {message_suffix}"
+        elif "data" in deployment_desc.config.roles:
```

Comment:
I don't think we should have this. This would prevent any dedicated CM cluster to ever run the below logic
I believe we should check on the existence of "data" in the peer-cluster-apps object given we have all visibility over the fleet

Reply:
adjusted.